### PR TITLE
Add release/snapshot building

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Add `make release-manager-release` and `make release-manager-snapshot` targets that cross-build fleet-server. Output is placed into `build/distributions`.

## Why is it important?

So final build artifacts can be built.

## Checklist

- [X] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Logs

```
GOOS=darwin GOARCH=amd64 go build -ldflags="-w -s -X main.Version=8.0.0-SNAPSHOT" -o build/binaries/fleet-server-8.0.0-SNAPSHOT-darwin-x86_64/fleet-server .
GOOS=linux GOARCH=386 go build -ldflags="-w -s -X main.Version=8.0.0-SNAPSHOT" -o build/binaries/fleet-server-8.0.0-SNAPSHOT-linux-x86/fleet-server .
GOOS=linux GOARCH=amd64 go build -ldflags="-w -s -X main.Version=8.0.0-SNAPSHOT" -o build/binaries/fleet-server-8.0.0-SNAPSHOT-linux-x86_64/fleet-server .
GOOS=linux GOARCH=arm64 go build -ldflags="-w -s -X main.Version=8.0.0-SNAPSHOT" -o build/binaries/fleet-server-8.0.0-SNAPSHOT-linux-arm64/fleet-server .
GOOS=windows GOARCH=386 go build -ldflags="-w -s -X main.Version=8.0.0-SNAPSHOT" -o build/binaries/fleet-server-8.0.0-SNAPSHOT-windows-x86/fleet-server .
GOOS=windows GOARCH=amd64 go build -ldflags="-w -s -X main.Version=8.0.0-SNAPSHOT" -o build/binaries/fleet-server-8.0.0-SNAPSHOT-windows-x86_64/fleet-server .
```

### CODE REVIEW

File: .go-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.14.12
```
Comment (ruflin) [inline .go-version:1]: Nit: Should we go to 1.15.?

File: Makefile [hunk 2]
```diff
@@ -14,33 +21,16 @@ help: ## - Show help message
...
-	@printf "${CMD_COLOR_ON} Build rpm\n${CMD_COLOR_OFF}"
-	@${DOCKER_BUILD} --ssh default  --target rpm -o ./  .
-
-.PHONY: image
-image:	## - Build the elastic fleet docker images
-	@printf "${CMD_COLOR_ON} Build the elastic fleet docker image\n${CMD_COLOR_OFF}"
-	${DOCKER_BUILD} --ssh default --target fleet -t fleet .
-
-.PHONY: run
-run: image ## - Run the smallest and secured golang docker image based on scratch
```
Comment (aleksmaus) [inline Makefile:28]: nit: are we removing docker image build option?

**Review by ruflin — APPROVED** [review]
I did run both commands locally and tried to execute the darwin SNAPSHOT binary and seems to do what it should do.

**Review by aleksmaus — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (blakerouse) [reply Makefile:28]: Yes, we don't need to build in docker. Its simple cross architecture building, which golang does natively.

File: .go-version (hunk 1 above)
Comment (blakerouse) [reply .go-version:1]: This is matching beats. I think we should just map to beats, when beats moves we move.

Comment (ruflin) [reply .go-version:1]: I think the nice part of being a separate binary is that we are not tied to what Beats has. I would rather tie it to the version the Elastic Agent uses if needed.

**Pull request merged**